### PR TITLE
Next Auth

### PR DESCRIPTION
**Pull request by UZ9**
## PR Title
Issue Number(s): #34 

What does this PR change and why?
### Checklist
- [x] Requirements have been implemented
- [x] Acceptance criteria is met
- [x] Database schema [docs](https://www.notion.so/gtbitsofgood/Database-Schema-Docs-a841a38d02db4428a7cdc85dcbb8a35c?pvs=4) have been updated or are not necessary
- [x] Code follows design and style guidelines
- [x] Commits follow guidelines (concise, squashed, etc)
- [x] Relevant reviewers (EM) have been assigned to this PR

### Critical Changes
- Database change / migration to run
- Environment config change
- Breaking API change

### How To Test
The steps to test the functionality of your ticket. Include edge cases and testing of error handling, if applicable.

### CODE REVIEW

File: src/pages/api/auth/[...nextauth].js [hunk 1]
```diff
@@ -1,10 +1,67 @@
+import { MongoDBAdapter } from "@next-auth/mongodb-adapter";
 import NextAuth from "next-auth";
+import dbConnect, { DB_INFO } from "../../../../server/db/dbConnect";
+import GoogleProvider from "next-auth/providers/google";
 
+let cachedPromise;
+
+import { MongoClient } from "mongodb";
```
Comment (afazio1) [inline src/pages/api/auth/[...nextauth].js:8]: remove if not using

**Review by afazio1 — COMMENTED** [review]
looks good so far! good work getting the svgs to render properly. Don't forget to add user db docs on notion when ur done with the schema 👍

File: server/db/dbConnect.js [hunk 2]
```diff
@@ -2,6 +2,12 @@ import mongoose from "mongoose";
 
 const DB_URL = process.env.DB_URL;
 const DB_NAME = process.env.DB_NAME;
+const DB_CONNECTION_STRING = `${DB_URL}${DB_NAME}?retryWrites=true&w=majority`
+const DB_OPTS = {
+  bufferCommands: false,
+};
+
+export const DB_INFO = { DB_URL, DB_NAME, DB_CONNECTION_STRING, DB_OPTS };
```
Comment (afazio1) [inline server/db/dbConnect.js]: maybe export the `DB_INFO` variables separately and on an as needed basis. I don't think we need `DB_INFO` as another layer of abstraction for env vars
```js
export const DB_URL
export const DB_NAME
...
```

File: src/pages/api/auth/[...nextauth].js [hunk 3]
```diff
@@ -1,10 +1,100 @@
...
+import dbConnect, { DB_INFO } from "../../../../server/db/dbConnect";
+import GoogleProvider from "next-auth/providers/google";
 
+let cachedPromise;
+
+import { MongoClient, ObjectId } from "mongodb";
+import User from "../../../../server/db/models/User";
+import { consts } from "@/utils/consts";
+
+const client = new MongoClient(DB_INFO.DB_CONNECTION_STRING);
```
Comment (afazio1) [inline src/pages/api/auth/[...nextauth].js]: we only use connection string here, so there's not much of a need to pull in all of `DB_INFO`

File: src/pages/api/auth/[...nextauth].js [hunk 4]
```diff
@@ -1,10 +1,100 @@
...
+    error: "/login",
+  },
+  events: {
+    createUser: async (message) => {
+      await dbConnect();
+
+      const _id = new ObjectId(message.user.id);
+
+      // TODO: Find a way of avoiding unnecessary deletion calls
+      await User.deleteOne({ _id });
```
Comment (afazio1) [inline src/pages/api/auth/[...nextauth].js:51]: this is not too much of a problem tbh. the operation only happens when new users are created, so not very often

File: src/utils/consts.js [hunk 5]
```diff
@@ -22,6 +22,7 @@ const consts = {
   concernArray: ["No Concern", "Some Concern", "High Concern"],
   locationArray: ["Facility 1", "Facility 2", "Placed"],
   roleArray: ["Service", "Companion"],
+  userRoleArray: ["Admin", "Instructor", "Volunteer", "Inactive"],
```
Comment (afazio1) [inline src/utils/consts.js]: leadership decided to change this to 
`["Admin", "Instructor", "Volunteer/Recipient", "Inactive"]`

File: src/pages/api/auth/[...nextauth].js [hunk 6]
```diff
@@ -1,10 +1,100 @@
...
+
+      const newUserRecord = {
+        _id,
+        name: message.user.name,
+        email: message.user.email,
+        image: message.user.image,
+        role: "Volunteer" // TODO: Default string shouldn't be hardcoded and instead moved somewhere else
```
Comment (afazio1) [inline src/pages/api/auth/[...nextauth].js]: make this "Volunteer/Recipient"
- we will probably want to give them a default role of "Inactive" until an admin activates them, but for testing purposes let's leave the default as this

**Review by afazio1 — CHANGES_REQUESTED** [review]
good job! a few minor fixes and should be good to go

File: server/db/dbConnect.js (hunk 2 above)
Comment (UZ9) [reply server/db/dbConnect.js]: Done 👍

File: src/pages/api/auth/[...nextauth].js (hunk 3 above)
Comment (UZ9) [reply src/pages/api/auth/[...nextauth].js]: Done 👍  (same as above)

File: src/pages/api/auth/[...nextauth].js (hunk 4 above)
Comment (UZ9) [reply src/pages/api/auth/[...nextauth].js:51]: Should I leave the TODO in case we ever come back to it much later?

File: src/utils/consts.js (hunk 5 above)
Comment (UZ9) [reply src/utils/consts.js]: 👍 Done

File: src/pages/api/auth/[...nextauth].js (hunk 6 above)
Comment (UZ9) [reply src/pages/api/auth/[...nextauth].js]: 👍 Done

File: src/pages/api/auth/[...nextauth].js (hunk 4 above)
Comment (afazio1) [reply src/pages/api/auth/[...nextauth].js:51]: sure

**Review by afazio1 — APPROVED** [review]
